packages/sources/s3reader/src/transport/csv.ts

```diff
@@ -0,0 +1,131 @@
+import { S3Client } from '@aws-sdk/client-s3'
+import { getFileFromS3 } from './utils'
+import { parse } from 'csv-parse/sync'
+
+const logger = makeLogger('S3PollerTransport')
+
+export type TransportTypes = BaseEndpointTypes
+type RequestParams = typeof inputParameters.validated
+
+export class S3PollerTransport extends SubscriptionTransport<TransportTypes> {
+  settings!: TransportTypes['Settings']
+  requester!: Requester
+  endpointName!: string
+  s3Client!: S3Client
+
+  async initialize(
+    dependencies: TransportDependencies<TransportTypes>,
+    adapterSettings: TransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+    this.settings = adapterSettings
+    this.requester = dependencies.requester
+    this.endpointName = endpointName
+    this.s3Client = new S3Client({})
+  }
+
+  async backgroundHandler(context: EndpointContext<TransportTypes>, entries: RequestParams[]) {
+    await Promise.all(entries.map(async (param) => this.handleRequest(param)))
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest(param: RequestParams) {
+    let response: AdapterResponse<BaseEndpointTypes['Response']>
+    try {
+      response = await this._handleRequest(param)
+    } catch (e) {
+      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
+      logger.error(e, errorMessage)
+      response = {
+        statusCode: 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: 0,
+          providerDataReceivedUnixMs: 0,
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: param, response }])
+  }
+
+  async _handleRequest(param: RequestParams): Promise<AdapterResponse<TransportTypes['Response']>> {
+    const { bucket, key, headerRow, resultColumn, matcherColumn, matcherValue } = param
+    const providerDataRequestedUnixMs = Date.now()
+    const csvFileAsStr = await getFileFromS3(this.s3Client, bucket, key)
+    const answer = this.parseCSV(csvFileAsStr, headerRow, matcherColumn, matcherValue, resultColumn)
+
+    return {
+      result: answer,
+      data: {
+        result: answer,
+      },
+      statusCode: 200,
+      timestamps: {
+        providerDataRequestedUnixMs,
+        providerDataReceivedUnixMs: Date.now(),
+        providerIndicatedTimeUnixMs: undefined,
+      },
+    }
+  }
+
+  // csvFileAsStr: CSV file as string as received from S3
+  // headerRow: 1-indexed row of the CSV file to use as the header row
+  // matcherColumn: field to match with `matcherValue` to find the answer row
+  // matcherValue: value of field `matcherColumn` used to find the answer row
+  // resultColumn: header field containing the answer in matcher row
+  parseCSV(
```

Comment:
Maybe pedantic, but I am not sure this is the best name for this function, as it returns a result, it isn't just about parsing/transforming the CSV. Maybe parseAndExtractCSV, extractFromCSV or fetchCSVCells?

Reply:
Renamed to `findValueInCSV`